libs/common/src/auth/services/webauthn-login/webauthn-login.service.spec.ts

```diff
@@ -0,0 +1,379 @@
+  // assert on the value in tests
+  private prfKeyArrayBuffer: ArrayBuffer = Utils.hexStringToArrayBuffer(
+    "1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+  );
+
+  getClientExtensionResults(): any {
+    return {
+      prf: {
+        results: {
+          first: this.prfKeyArrayBuffer,
+        },
+      },
+    };
+  }
+
+  static isConditionalMediationAvailable(): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+
+  static isUserVerifyingPlatformAuthenticatorAvailable(): Promise<boolean> {
+    return Promise.resolve(false);
+  }
+}
+
+// We must do this to make the mocked classes available for all the
+// assertCredential(...) tests.
+global.PublicKeyCredential = MockPublicKeyCredential;
+global.AuthenticatorAssertionResponse = MockAuthenticatorAssertionResponse;
+
+describe("WebAuthnLoginService", () => {
+  let webAuthnLoginService: WebAuthnLoginService;
+
+  const webAuthnLoginApiService = mock<WebAuthnLoginApiServiceAbstraction>();
+  const authService = mock<AuthService>();
+  const configService = mock<ConfigServiceAbstraction>();
+  const navigatorCredentials = mock<CredentialsContainer>();
+  const logService = mock<LogService>();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    // Restore global after all tests are done
+    global.PublicKeyCredential = originalPublicKeyCredential;
+    global.AuthenticatorAssertionResponse = originalAuthenticatorAssertionResponse;
+  });
+
+  function setup(enabled: boolean): WebAuthnLoginService {
+    configService.getFeatureFlag$.mockReturnValue(of(enabled));
+    return new WebAuthnLoginService(
+      webAuthnLoginApiService,
+      authService,
+      configService,
+      navigatorCredentials,
+      logService
+    );
+  }
+
+  it("instantiates", () => {
+    webAuthnLoginService = setup(true);
+    expect(webAuthnLoginService).not.toBeFalsy();
+  });
+
+  describe("enabled$", () => {
+    it("should emit true when feature flag for PasswordlessLogin is enabled", (done) => {
+      // Arrange
+      const webAuthnLoginService = setup(true);
+
+      // Act & Assert
+      webAuthnLoginService.enabled$.subscribe({
+        next: (enabled) => {
+          expect(enabled).toBe(true);
+          done();
+        },
+        error: done.fail,
+      });
+    });
+
+    it("should emit false when feature flag for PasswordlessLogin is disabled", (done) => {
+      // Arrange
+      const webAuthnLoginService = setup(false);
+
+      // Act & Assert
+      webAuthnLoginService.enabled$.subscribe({
+        next: (enabled) => {
+          expect(enabled).toBe(false);
+          done();
+        },
+        error: done.fail,
+      });
+    });
+  });
+
+  describe("getCredentialAssertionOptions()", () => {
+    it("should get credential assertion options and return a WebAuthnLoginAssertionOptionsView", async () => {
+      // Arrange
+      const webAuthnLoginService = setup(true);
```

Comment:
suggestion(non-blocking): would you be open to changing the setup method to make it a bit easier to read? `setup(true)` doesn't say very much and requires that you go and read the implementation

Something like
```suggestion
      const webAuthnLoginService = createService({ featureFlag: true });
```